From 1910c01834dd41015993fc3b5646dd76d01c164f Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 4 Feb 2016 14:53:49 +0100
Subject: [PATCH] S2: Add defined state for several ctrl gpios

Set several gpios in a defined state. This is needed to avoid silly
effects by floating IOs on peripherals

Ticket #927
---
 board/freescale/mx6sxscore/mx6sxscore.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/board/freescale/mx6sxscore/mx6sxscore.c b/board/freescale/mx6sxscore/mx6sxscore.c
index d6833d6..ae48cc6 100644
--- a/board/freescale/mx6sxscore/mx6sxscore.c
+++ b/board/freescale/mx6sxscore/mx6sxscore.c
@@ -319,6 +319,10 @@ static iomux_v3_cfg_t const gpio_pads[] = {
     MX6SX_PAD_GPIO1_IO02__GPIO1_IO_2 | MUX_PAD_CTRL(NO_PAD_CTRL), // CAN Standby
     MX6SX_PAD_KEY_COL2__GPIO2_IO_12 | MUX_PAD_CTRL(NO_PAD_CTRL), // 5PE
     MX6SX_PAD_LCD1_DATA10__GPIO3_IO_11 | MUX_PAD_CTRL(NO_PAD_CTRL), // KILL_MAIN_PWR
+    MX6SX_PAD_LCD1_HSYNC__GPIO3_IO_26 | MUX_PAD_CTRL(NO_PAD_CTRL), // WIFI_PWR_E
+    MX6SX_PAD_LCD1_DATA08__GPIO3_IO_9 | MUX_PAD_CTRL(NO_PAD_CTRL), // GNSS_PWR_E
+    MX6SX_PAD_LCD1_DATA14__GPIO3_IO_15 | MUX_PAD_CTRL(NO_PAD_CTRL), // GSM_PWR_E
+    MX6SX_PAD_GPIO1_IO09__GPIO1_IO_9 | MUX_PAD_CTRL(NO_PAD_CTRL), // GSM_REs
 #endif // CONFIG_SITEC_S2
 };
 
@@ -333,6 +337,10 @@ static void setup_gpio(void)
     gpio_direction_output(IMX_GPIO_NR(1, 2), 0);
     gpio_direction_output(IMX_GPIO_NR(2, 12), 1);
     gpio_direction_output(IMX_GPIO_NR(3, 11), 0);
+    gpio_direction_output(IMX_GPIO_NR(3, 26), 0);
+    gpio_direction_output(IMX_GPIO_NR(3, 9), 0);
+    gpio_direction_output(IMX_GPIO_NR(3, 15), 0);
+    gpio_direction_output(IMX_GPIO_NR(1, 9), 1);
 #endif // CONFIG_SITEC_S2
 }
 
-- 
1.9.1

